src/pygama/lgdo/lh5_store.py

```diff
@@ -622,8 +622,9 @@ def write_object(self,
 
         # scalars
         elif isinstance(obj, Scalar):
-            if name in group:
-                log.debug(f"overwriting '{name}' in '{group}'")
+            if wo_mode == 'o' and name in group:
```

Comment:
I also feel like if `name` is in `group` and `wo_mode` is `'w'` then we should emit an error here. @iguinn do you agree?

Also `wo_mode = 'a' should be okay for overwriting scalars, according to the `write_object` docstring.

Reply:
I made this change.

Reply:
Ok, sounds good. I guess the other case is 'of' but in that case I think no check is needed since the file's overwritten so it shouldn't be possible to run into this? Either way, raising an error ourselves is a good idea, as before it was raising a somewhat confusing h5py error.